distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/KadAction.java

```diff
@@ -0,0 +1,383 @@
+            case RESOURCE:
+                try {
+                    return StringResource.parseString(payload, RESOURCE_SEPARATOR).isValid();
+                }
+                catch (IllegalArgumentException e){
+                    return false;
+                }
+            default:
+                return false;
+        }
+
+    }
+
+    /**
+     * Gets a message with correct formatted action command ready to be sent.
+     *
+     * @return A SMSMessage containing the formatted action command ready to be sent.
+     */
+    @Override
+    public SMSMessage toMessage() {
+        StringBuilder body = new StringBuilder();
+
+        body.append(actionType.getCode()).append(addPadding(operationId,OPERATION_ID_LENGTH))
+                .append(addPadding(currentPart,CURRENT_PART_LENGTH)).append(addPadding(totalParts,TOTAL_PARTS_LENGTH))
+                .append(payloadType.getCode()).append(payload);
+
+        return new SMSMessage(actionPeer, body.toString());
+    }
+
+    /**
+     * Gets the action type from the enum.
+     *
+     * @return The type of the action.
+     */
+    public ActionType getActionType() {
+        return actionType;
+    }
+
+    /**
+     * Gets the ID which identifies the action.
+     *
+     * @return The operation ID.
+     */
+    public int getOperationId() {
+        return operationId;
+    }
+
+    /**
+     * Gets the k part of the current message.
+     *
+     * @return The currentPart k  of the message.
+     */
+    public int getCurrentPart() {
+        return currentPart;
+    }
+
+    /**
+     * Gets the number of the total messages expected .
+     *
+     * @return The total number of messages.
+     */
+    public int getTotalParts() {
+        return totalParts;
+    }
+
+    /**
+     * Gets the payload type from the enum.
+     *
+     * @return The type of the payload.
+     */
+    public PayloadType getPayloadType() {
+        return payloadType;
+    }
+
+    /**
+     * Gets the SMSPeer which is sending or receiving the action message.
+     *
+     * @return SMSPeer the peer sending or receiving the action message.
+     */
+    @Override
+    public SMSPeer getPeer() {
+        return actionPeer;
+    }
+
+    /**
+     * Gets a string containing the value of the payload
+     *
+     * @return A string containing the value of the payload.
+     */
+    @Override
+    public String getPayload() {
+        return payload;
+    }
+
+    /**
+     * Converts an integer to a string and adds to it the padding character.
+     *
+     * @param intToPadd The integer that need the padding.
+     * @param length The wanted length for the string.
+     * @return String with the correct length.
+     */
+    static String addPadding(int intToPadd,int length) {
+        String stringToPadd=Integer.toString(intToPadd);
+        while(stringToPadd.length()<length)
+        {
+            stringToPadd=PARSING_CHARACTER+stringToPadd;
+        }
+        return stringToPadd;
+    }
```

Comment:
What's gonna happen if someone gives a negative number as length?
It would return the same string, so it would just convert the int value into a String value, right?
Are you sure to leave this method like this?
If so, you might consider to be more specific in specifications... maybe telling that, if length is NOT greater than 0, it would just convert the int value into a String value.

Reply:
You're right , I'll change specifications. By the way is length not greater than stringToPadd.length I think

Reply:
Yeah, you are right. It happends the same thing if length is <= stringToPadd.length() ...